src/routes/oauth/oauthUser.ts

```diff
@@ -17,4 +21,24 @@ export default class extends Route {
 		return this.client.dashboardUsers.add(user);
 	}
 
+	@authenticated
+	@ratelimit(2, 1000 * 60 * 5, true)
+	public async post(request: ApiRequest, response: ApiResponse) {
+		const requestBody = request.body as Record<string, string>;
+		if (!requestBody.action) {
+			return response.error(400);
+		}
+
+		if (requestBody.action === 'SYNC_USER') {
+			const user = await this.api(request.auth!.token)
+				.catch(err => {
+					this.client.emit(Events.Wtf, err);
+				});
+
+			return user ? response.json({ user }) : response.error(500);
+		}
+
+		return response.error(400);
```

Comment:
This could all be changed to check if the action is not `SYNC_USER`, and error accordingly

Reply:
It's coded to be easy to add more actions

Reply:
This can also error because the token expires in a week, we need to save the refresh and the expire at values somewhere.

My idea was making a table for that, it's not suitable in user settings due to it making the db and RAM usage much bigger.